### PR TITLE
[miniupnpc] Update to 2.3.7

### PR DESCRIPTION
**Pull request by MonicaLiu0311**
Update `miniupnpc` to 2.3.7. No feature needs to be tested, the usage test succesfully on `x64-windows`(header file found):
```
miniupnpc provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(miniupnpc CONFIG REQUIRED)
  target_link_libraries(main PRIVATE miniupnpc::miniupnpc)
```

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version.~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [ ] ~Any patches that are no longer applied are deleted from the port's directory.~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/miniupnpc/portfile.cmake [hunk 1]
```diff
@@ -32,11 +32,13 @@ if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
...
-    )
-    vcpkg_copy_tool_dependencies("${CURRENT_PACKAGES_DIR}/tools/${PORT}")
-    file(REMOVE ${DEBUG_TOOLS} ${RELEASE_TOOLS})
+    if(DEBUG_TOOLS OR RELEASE_TOOLS)
+        file(
+            INSTALL ${RELEASE_TOOLS}
+            DESTINATION "${CURRENT_PACKAGES_DIR}/tools/${PORT}"
+        )
+        vcpkg_copy_tool_dependencies("${CURRENT_PACKAGES_DIR}/tools/${PORT}")
+        file(REMOVE_RECURSE ${DEBUG_TOOLS} ${RELEASE_TOOLS})
```
Comment (jimwang118) [inline ports/miniupnpc/portfile.cmake:41]:
```suggestion
        vcpkg_copy_tools(TOOL_NAMES ${RELEASE_TOOLS} AUTO_CLEAN)

```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Review by jimwang118 — APPROVED** [review]

File: ports/miniupnpc/portfile.cmake [hunk 2]
```diff
@@ -2,8 +2,8 @@ string(REPLACE "." "_" MINIUPNPC_VERSION "${VERSION}")
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO miniupnp/miniupnp
-    REF "miniupnpc_${MINIUPNPC_VERSION}"
-    SHA512 2f1fcd2d2820f649b832e1e0d6b826a812ee84f634460acb3b4af15338488fe3889bb8929c38a91a4f9786a5232177df479d874a388230723d27ae708d3b2592
+    REF "miniupnpd_${MINIUPNPC_VERSION}"
+    SHA512 f461b57446d9086e8fc2caf6ef202a78d204b2580d6096aca63234fda53302cad03840b14cc0d8b68e9033049b749be46e35c3737cc6cbcd2ad30c59b737a390
```
[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/miniupnpc/portfile.cmake:6]:
~~~
miniupnpc/ : MiniUPnP client - an UPnP IGD control point
miniupnpd/ : MiniUPnP daemon - an implementation of a UPnP IGD
                               + NAT-PMP / PCP gateway

[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/miniupnpc/portfile.cmake:6]: @BillyONeal How to get rid of 2.3.7 which is not miniupnpc 2.3.7?

[the remark this answers is not in the document]
Comment (MonicaLiu0311) [reply ports/miniupnpc/portfile.cmake:6]: 😭 Maybe #43233 will be fixed.

**Pull request merged**